backend/corpora/common/entities/tiledb_data.py

```diff
@@ -0,0 +1,408 @@
+            # internal TileDB schema
+            "visibility",
+            "revision_of",
+            "owner",
+        ],
+        "datasets": [
+            "X_approximate_distribution",
+            "X_normalization_assay",
+            "cell_count",
+            "cell_type",
+            "development_stage",
+            "disease",
+            "ethnicity",
+            "is_primary_data",
+            "name",
+            "organism",
+            "sex",
+            "tissue",
+        ]
+    }
+       
+    attrs_to_parse = {
+        "collections":   [
+            "links",
+            "datasets",
+            "publisher_metadata"
+        ],
+        "datasets": [
+            "assay",
+            "cell_type",
+            "development_stage",
+            "disease",
+            "ethnicity",
+            "organism",
+            "sex",
+            "tissue"
+        ]
+    }
+  
+    @staticmethod
+    def new_id():
+        return uuid.uuid4().hex
+
+    # TODO: figure out actual list and dict support in TileDB
+    @staticmethod
+    def parse_stored_data(data: dict, array: str) -> dict:
+        for a in Utils.attrs_to_parse[array]:
+            if a in data:
+                data[a] = ast.literal_eval(data[a])
+        return data
+
+    @staticmethod
+    def pack_input_data(data: dict, array: str) -> dict:
+        for a in Utils.attrs_to_parse[array]:
+            if a in data:
+                data[a] = str(data[a])
+        return data
+
+
+
+
+class TileDBData():
+    @staticmethod
+    def init_db(location):
+        """FOR TESTING PURPOSES, create a local TileDB group and arrays according to our schema."""
+        # create group
+        if os.path.exists(location):
+            shutil.rmtree(location)
+
+        tiledb.group_create(location)
+
+        # create collections array
+        # TODO: figure out ideal domain and tile
+        # TODO: should we use more than one dimension?
+        dim1 = tiledb.Dim(name="id", domain=(None, None), tile=2, dtype="S0")
+        dom = tiledb.Domain(dim1)
+
+        a1 = tiledb.Attr(name="owner", dtype="U1")
+        a2 = tiledb.Attr(name="visibility", dtype="U1") # DELETED, PRIVATE, PUBLIC
+        a3 = tiledb.Attr(name="name", dtype="U1")
+        a4 = tiledb.Attr(name="description", dtype="U1")
+        a5 = tiledb.Attr(name="contact_name", dtype="U1")
+        a6 = tiledb.Attr(name="contact_email", dtype="U1")
+        a7 = tiledb.Attr(name="links", dtype="U1")
+        a8 = tiledb.Attr(name="datasets", dtype="U1")
+        a9 = tiledb.Attr(name="revision_of", dtype="U1")
+        a10 = tiledb.Attr(name="curator_name", dtype="U1")
+        a11 = tiledb.Attr(name="created_at", dtype=np.float32)
+        a12 = tiledb.Attr(name="updated_at", dtype=np.float32)
+        a13 = tiledb.Attr(name="publisher_metadata", dtype="U1")
+
+        schema = tiledb.ArraySchema(domain=dom, sparse=True, attrs=[a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13])
```

Comment:
you should not use `sparse=True` in this context

Reply:
But don't you need a sparse array to handle string (uuid) dimensions?

Reply:
I don't think so?

Reply:
Docs may be wrong but I'm going off this: https://docs.tiledb.com/main/how-to/arrays/creating-arrays/creating-dimensions